jwst/transforms/models.py

```diff
@@ -1406,25 +1444,44 @@ def evaluate(self, x, y, wavelength, order):
         reference file and fwcpos from the input image.
 
         """
-        if wavelength < 0:
+        if (wavelength < 0).any():
             raise ValueError("Wavelength should be greater than zero")
         try:
-            iorder = self._order_mapping[int(order)]
+            iorder = self._order_mapping[int(order.flatten()[0])]
         except KeyError:
             raise ValueError("Specified order is not available")
 
         t = self.lmodels[iorder](wavelength)
-        # use that t to compute the dx and dy
-        dx = self.xmodels[iorder][0](x, y) + t * self.xmodels[iorder][1](x, y)
-        dy = self.ymodels[iorder][0](x, y) + t * self.ymodels[iorder][1](x, y)
-        # rotate by theta
+        xmodel = self.xmodels[iorder]
+        ymodel = self.ymodels[iorder]
+        lmodel = self.lmodels[iorder]
+
+        dx = xmodel[0](x, y) + t * xmodel[1](x, y)
+        dy = ymodel[0](x, y) + t * ymodel[1](x, y)
+
+        ## use that t to compute the dx and dy
```

Comment:
Are all of these commented blocks new code that will be activated in the future?

Reply:
Everything is activated now, I will clean the code.